### PR TITLE
feat: Iterator for single shard in the tree

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR implements the iterator ShardIter to scan a single shard in the merge tree.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/mito2/src/memtable/merge_tree/tree.rs [hunk 1]
```diff
@@ -235,6 +278,132 @@ struct TreeParts {
...
+    /// Converts current [RecordBatch] to [Batch].
+    fn convert_current_record_batch(
+        &self,
+        metadata: &RegionMetadataRef,
+        projection: &HashSet<ColumnId>,
+        primary_key: &[u8],
+    ) -> Result<Batch> {
+        let data_batch = self.current.as_ref().unwrap();
+        let offset = data_batch.range().start;
+        let length = data_batch.range().end - offset;
```
Comment (v0y4g3r) [inline src/mito2/src/memtable/merge_tree/tree.rs]:
```suggestion
        let length = data_batch.range().len();
```

File: src/mito2/src/memtable/merge_tree/tree.rs [hunk 2]
```diff
@@ -235,6 +278,132 @@ struct TreeParts {
...
+    fn is_valid(&self) -> bool {
+        self.current.is_some()
+    }
+
+    fn current_pk_index(&self) -> PkIndex {
+        self.current.as_ref().unwrap().pk_index()
+    }
+
+    /// Converts current [RecordBatch] to [Batch].
+    fn convert_current_record_batch(
```
Comment (v0y4g3r) [inline src/mito2/src/memtable/merge_tree/tree.rs:364]: This PR provides a `DataBatch::as_record_batch` method:
https://github.com/evenyag/greptimedb/pull/15/files#diff-5066fcdd3b887926278b1155cfbf005cbca27e9699e36292abcc55ecf80110e2R61-R63

Comment (evenyag) [reply src/mito2/src/memtable/merge_tree/tree.rs:364]: Slicing the record batch is a bit wasteful as it clones the underlying `Vec<ArrayRef>`. The data iter also has this issue as it has to clone the record batch for each pk index.

File: src/mito2/src/memtable/merge_tree/tree.rs (hunk 1 above)
Comment (evenyag) [reply src/mito2/src/memtable/merge_tree/tree.rs]: Addressed in https://github.com/evenyag/greptimedb/pull/14/commits/252f7da1de2703ae1bb2ab5318b2557e3effc46a

**Pull request merged**